build/commands/lib/util.js

```diff
@@ -633,8 +637,9 @@ const util = {
       if (prevBuildArgs)
         fs.removeSync(buildArgsFile)
 
-      util.run('gn', ['gen', config.outputDir, '--args="' + buildArgsStr + '"'], options)
+      util.run('gn', ['gen', config.outputDir, '--args="' + buildArgsStr + '"', config.extraGnGenOpts], options)
```

Comment:
the right way is to store it as an array in `.npmrc` and extend the `args` array properly here.
`brave_extra_gn_gen_opts=['--ide=vs', '--sln=brave', '--filters=//brave/*']`

Reply:
You mean like:
```
extra_gn_gen_opts[]=--ide=vs
extra_gn_gen_opts[]=--sln=brave
extra_gn_gen_opts[]=--filters=//brave/*
```
right? That's what I was trying to do originally. Here's what
```
const list = run(npmCommand, ['config', 'list', '--json', '--userconfig=' + path.join(rootDir, '.npmrc')])
NpmConfig = JSON.parse(list.stdout.toString())
console.log(NpmConfig)
```
yields:
```
...
'extra-gn-gen-opts': '--ide=vs\n\n--sln=brave\n\n--filters=//brave/*',
...
```
whereas
```
npm config list --json --userconfig .npmrc
```
gives:
```
...
"extra_gn_gen_opts": [
  "--ide=vs",
  "--sln=brave",
  "--filters=//brave/*"
],
...
```
I saw [here](https://docs.npmjs.com/cli/v8/configuring-npm/npmrc) that npmrc should support arrays. I might be missing something.

Reply:
seems like `--userconfig` makes npm do weird stuff when `path.join(rootDir, '.npmrc')` doesn't exist. I have no idea what's going on in this case.

if you add a check to not add `--userconfig` if the path doesn't exist, then everything seems to normalize.

Reply:
Seems a bit hacky to me. Actually, would we gain any benefit from storing these options in an array? I mean, other than having them temporarily better structured until we pass them to `gn gen`?

Reply:
> would we gain any benefit from storing these options in an array?

we don't have to store them in an array, but I assume it's better to pass them as separate items to `spawnSync`  in case they will contain any chars that require escaping.
Or we can assume that no escaping is needed, then we can pass it as is.

Reply:
How about passing it as is for now and getting back to the workaround above if someone runs into an issue?